### PR TITLE
add test and prevent null values on task name setup

### PR DESCRIPTION
**Pull request by CollinBeczak**
Resolves: https://github.com/maproulette/maproulette3/issues/1761

Also added test for confirmation and to prevent regressions:
Screenshot 2024-08-29 at 2 47 57 PM

Test results before the change:
Screenshot 2024-09-01 at 1 52 46 PM

Key takeaways:

### CODE REVIEW

File: app/org/maproulette/provider/ChallengeProvider.scala [hunk 1]
```diff
@@ -282,85 +282,89 @@ class ChallengeProvider @Inject() (
...
+  def taskNameFromJsValue(value: JsValue, challenge: Challenge): String = {
     // Use field/property specified by challenge, if available. Otherwise look
-    // for commonly used id fields/properties
-    if (!challenge.extra.osmIdProperty.getOrElse("").isEmpty) {
-      return featureOSMId(value, challenge) match {
-        case Some(osmId) => osmId
-        case None        => UUID.randomUUID().toString // task does not contain id property
-      }
+    // for commonly used ID fields/properties
+    if (challenge.extra.osmIdProperty.exists(_.nonEmpty)) {
```
Comment (ljdelight) [inline app/org/maproulette/provider/ChallengeProvider.scala:340]: Nice improvement

**Review by ljdelight — APPROVED** [review]
lgtm
